crates/hf_model_downloader/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! `HuggingFace` Model Downloader
+//!
+//! This crate provides reusable functionality for downloading models from `HuggingFace` Hub.
+//! It handles authentication, caching, and supports various repository types (models, datasets, spaces).
+
+use hf_hub::{
+    Repo, RepoType,
+    api::{
+        RepoInfo,
+        tokio::{Api, ApiBuilder, ApiRepo},
+    },
+};
+use secrecy::{ExposeSecret, SecretString};
+use snafu::prelude::*;
+use std::path::PathBuf;
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("Failed to build HuggingFace API: {source}"))]
+    FailedToBuildApi {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+
+    #[snafu(display("Failed to download file '{file}' from HuggingFace: {source}"))]
+    FailedToDownloadFile {
+        file: String,
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+
+    #[snafu(display("Failed to get repository info: {source}"))]
+    FailedToGetRepoInfo {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+}
+
+/// Configuration for downloading models from `HuggingFace`
+#[derive(Debug, Clone)]
+pub struct DownloadConfig {
+    /// `HuggingFace` repository ID (e.g., "meta-llama/Llama-3.3-70B-Instruct")
+    pub repo_id: String,
```

Comment:
Most of this, and alot of the `impl DownloadConfig` is functionality already in `hf_hub::lib::Repo` and `hf_hub::lib::Api`? E.g.

```rust
pub struct Repo {
    repo_id: String,
    repo_type: RepoType,
    revision: String,
}
```